### PR TITLE
WELD-2628 Fix integer overflow when sorting alternatives

### PR DESCRIPTION
**Pull request by karlvr**
This PR fixes an integer overflow issue when sorting alternatives with a priority of `Integer.MIN_VALUE`, and includes a test proving the fix.

### CODE REVIEW

**Comment by manovotn** [comment]
ok to test

File: tests-arquillian/src/test/java/org/jboss/weld/tests/extensions/lifecycle/atd/minvaluepriority/AfterTypeDiscoveryObserver.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat, Inc., and individual contributors
```
Comment (manovotn) [inline tests-arquillian/src/test/java/org/jboss/weld/tests/extensions/lifecycle/atd/minvaluepriority/AfterTypeDiscoveryObserver.java:3]: Time travel? :-)
(other licenses are also affected)

**Comment by manovotn** [comment]
The CI failure looks unrelated, I'll take a closer look at that soon.

**Comment by manovotn** [comment]
So I am unclear whether the test is wrong or whether this is a bug. The test seems to start failing once WFLY core gets upgraded from `12.0.0.Beta2` to `Beta3`. This changes the underlying version of Undertow from `2.1.0.Final` to `2.1.1.Final`. 
My personal guess as to what issue may be related is https://issues.redhat.com/browse/UNDERTOW-1419

Our test expects that the session will be destroyed asynchronously, while we execute a "long running" task on conversation bean. This means the `@WebListener` and our `@PreDestroy` bean callback would both be invoked before the bean task is over.
On newer WFLY versions, this is no longer the case and the session doesn't get destroyed until a new request is sent some time after the timeout period. I am going to contact someone from Undertow team and ask them if what our test is doing is supported.

**Comment by manovotn** [comment]
Spec: 7.5 Session Timeouts:
> The session invalidation will not take effect until all servlets using that session have exited the service method.

Which would mean the test is probably incorrect but was somehow working up until now.

**Comment by karlvr** [comment]
I remain a bit confused by the test failures. Is there anything I can do at this stage? Shall I update those copyright years? I copied and pasted and didn't read :-D

**Comment by manovotn** [comment]
@karlvr sorry for the confusion, I'll fix the test today/tomorrow in a separate PR because at this point I know its faulty test (which just happened to work until recent change in WFLY).
But please change the copyright years. Otherwise this PR looks good!

**Comment by manovotn** [comment]
https://github.com/weld/core/pull/2000 should fix the failing test.
Retest this please

**Comment by manovotn** [comment]
Ah, tests are passing now!
@karlvr can you please fix the license years? Then we can merge this.

**Comment by manovotn** [comment]
Hmm the JDK 11 failure is a flaky that that I hoped we sorted out ([here](https://github.com/weld/core/commit/3cedc2628df3d3c6c2c1a410473dfd0118331b4d)), but apparently we didn't. It's CI-only, cannot reproduce that locally. Anyhow, this PR is fine, I will cherry pick these changes into master as well.
Retest this please.

**Pull request merged**